### PR TITLE
feat: add snackbar on URL copy

### PR DESCRIPTION
**Pull request by lcoenengithub**
**What is this?**

This adds a snackbar message at video URL copy

**Illustration**
Peek 2022-02-23 16-52 - video

### CODE REVIEW

File: packages/atlas/src/components/_video/VideoTileViewer/VideoTileViewer.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { useVideoTileSharedLogic } from '@/hooks/useVideoTileSharedLogic'
 import { copyToClipboard } from '@/utils/browser'
 import { formatDurationShort } from '@/utils/time'
 
+import { useSnackbar } from '../../../providers/snackbars/snackbar'
```
Comment (drillprop) [inline packages/atlas/src/components/_video/VideoTileViewer/VideoTileViewer.tsx]: you should use: 
```
import { useSnackbar } from '@/providers/snackbars'
```

**Review by drillprop — CHANGES_REQUESTED** [review]
Couple things:
- we should use `dev` as a base branch.
- snackbar should have an icon: https://www.figma.com/file/LF7LOMhOA46bRcYHWlw1Bq/My-videos-page?node-id=792%3A31187 
- snackbar should be displayed also when you copy the URL in VideoTilePublisher. You missed this.
- I would add a snack bar also in MembershipInfo. We're copying the address there.  Maybe we could create a hook(`useCopyToClipboard`) that will be handling copying to the clipboard and displaying the snack bar. @kdembler wdyt?

**Comment by drillprop** [comment]
Also, I have one more request. Maybe it's just oversight, but please add the link to the issue and to the design(if it's available) in your PRs.

**Comment by kdembler** [comment]
> I would add a snack bar also in MembershipInfo. We're copying the address there. Maybe we could create a hook(useCopyToClipboard) that will be handling copying to the clipboard and displaying the snack bar. @kdembler wdyt?

Very good point, agreed

**Comment by lcoenengithub** [comment]
@drillprop @kdembler thanks! I applied your points.

File: packages/atlas/src/providers/clipboard/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { useSnackbar } from '@/providers/snackbars'
+import { copyToClipboard } from '@/utils/browser'
+
+export const useClipboard = () => {
+  const { displaySnackbar } = useSnackbar()
+  return {
+    copyToClipboard: (text: string) => {
+      copyToClipboard(text)
+      displaySnackbar({ title: 'Video URL have been copied', iconType: 'info' })
+    },
```
Comment (drillprop) [inline packages/atlas/src/providers/clipboard/hooks.ts]: Firstly, I think, this should be "Video URL copied to clipboard". 

Secondly, when you click the copy button on the member page, you're copying the account address, not the video URL.
image
You need to add one more parameter to `copyToClipboard` function to display the correct information. Something like: 
```tsx
copyToClipboard: (text: string, snackbarText: string) => {
      copyToClipboard(text)
      displaySnackbar({ title: snackbarText, iconType: 'info' })
    },
```

**Review by drillprop — CHANGES_REQUESTED** [review]
Works well 👍🏻 , just one thing left

File: packages/atlas/src/providers/clipboard/hooks.ts (hunk 2 above)
Comment (lcoenengithub) [reply packages/atlas/src/providers/clipboard/hooks.ts]: Thanks @drillprops, I fixed it (I added some args with default values).

File: packages/atlas/src/providers/clipboard/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { SnackbarIconType, useSnackbar } from '@/providers/snackbars'
+import { copyToClipboard } from '@/utils/browser'
+
+export const useClipboard = () => {
+  const { displaySnackbar } = useSnackbar()
+  return {
+    copyToClipboard: (text: string, title = 'Video URL copied to clipboard', iconType: SnackbarIconType = 'info') => {
```
Comment (drillprop) [inline packages/atlas/src/providers/clipboard/hooks.ts]: hmm, I don't think we need to add `iconType` as a parameter - this should be always an informational snack bar.  I don't see the reason why we would change that. 

I also don't agree that we should set the default title.

File: packages/atlas/src/components/MembershipInfo/MembershipInfo.tsx [hunk 4]
```diff
@@ -46,6 +46,7 @@ export const MembershipInfo: React.FC<MembershipInfoProps> = ({
   editable,
   className,
 }) => {
+  const { copyToClipboard } = useClipboard()
```
Comment (drillprop) [inline packages/atlas/src/components/MembershipInfo/MembershipInfo.tsx:49]: we need to pass suitable snackbarText on line 57 i.e: `Account address copied to clipboard`

**Review by drillprop — CHANGES_REQUESTED** [review]

**Comment by drillprop** [comment]
@lcoenengithub I'm confused,  you requested my review, but I don't see any new commits or comments.  That was a misclick or did you forget to push your changes?

**Comment by lcoenengithub** [comment]
> @lcoenengithub I'm confused, you requested my review, but I don't see any new commits or comments. That was a misclick or did you forget to push your changes?

Hey! Sorry, I forgot to push indeed :)

File: packages/atlas/src/components/MembershipInfo/MembershipInfo.tsx [hunk 5]
```diff
@@ -46,14 +46,15 @@ export const MembershipInfo: React.FC<MembershipInfoProps> = ({
...
   const [copyButtonClicked, setCopyButtonClicked] = useState(false)
   const smMatch = useMediaMatch('sm')
 
   const handleCopyAddress = () => {
     if (!address) {
       return
     }
-    copyToClipboard(address)
+    copyToClipboard(address, 'Video URL copied to clipboard')
```
Comment (drillprop) [inline packages/atlas/src/components/MembershipInfo/MembershipInfo.tsx]:
```suggestion
    copyToClipboard(address, 'Account address copied to clipboard')
```

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/providers/clipboard/hooks.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { useSnackbar } from '@/providers/snackbars'
+import { copyToClipboard } from '@/utils/browser'
+
+export const useClipboard = () => {
```
Comment (kdembler) [inline packages/atlas/src/providers/clipboard/hooks.ts]: This isn't a provider though, right? We keep only context providers in `src/providers`, I think this hook makes more sense in `src/hooks`

Comment (lcoenengithub) [reply packages/atlas/src/providers/clipboard/hooks.ts]: Good point!

Comment (lcoenengithub) [reply packages/atlas/src/providers/clipboard/hooks.ts]: Good point, I moved it there!

**Review by drillprop — APPROVED** [review]

File: packages/atlas/src/hooks/clipboard/hooks.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { useSnackbar } from '@/providers/snackbars'
```
Comment (kdembler) [inline packages/atlas/src/hooks/clipboard/hooks.ts]: It's in `hooks/` now but it doesn't match how we name files in that directory, please take a look at other hooks, we should name appropriately and remove unnecessary directory

**Review by kdembler — APPROVED** [review]

**Pull request merged**